### DISCUSSION TITLE
Issue: override a selection background

### DISCUSSION THREAD

**Discussion by qbnil** [rose-pine/neovim#236] [Q&A]
### Neovim version (nvim -v)

0.9.5

### Terminal / multiplexer

kitty

### Describe the bug

i want to override the default white selection background in dawn variant in my setup(). I tried to look it up but didn't figure this out, will be glad to see a link at least where i can read on that :), thx

**Comment by mvllow** [comment]
From the readme:

```lua
{
    highlight_groups = {
        -- Comment = { fg = "foam" },
        -- VertSplit = { fg = "muted", bg = "muted" },
    },

    before_highlight = function(group, highlight, palette)
        -- Disable all undercurls
        -- if highlight.undercurl then
        --     highlight.undercurl = false
        -- end
        --
        -- Change palette colour
        -- if highlight.fg == palette.pine then
        --     highlight.fg = palette.foam
        -- end
    end,
}
```

If you want to modify a colour for dawn only:

```lua
{
    before_highlight = function(group, highlight, palette)
        if vim.o.background == "light" and group == "GroupName" then
             highlight.bg = palette.foam
        end
    end,
}
```

**Reply by qbnil** [reply]
Thx, gonna try that asap

**Reply by qbnil** [reply]
This doesn't seem to work tho, idk really know what i'm doing wrong, i try to change  the selection color to 'pine' for instance and it stays the same as it used to be

**Reply by qbnil** [reply]
i actually need to change the color of the current selection (i.e words that i personally selected in V-mode)

**Reply by mvllow** [reply]
Which groups have you tried? You may want to change `Visual`.

**Reply by qbnil** [reply]
i tried tweaking the before_highlight function

**Reply by qbnil** [reply]
do you mean trying the highlight.Visual?

**Reply by qbnil** [reply]
tried the highlight.visual, it only changes the text color when selecting over it

**Reply by qbnil** [reply]
figured it already, thx, had to add a Visual group to groups in my config and set the bg, appreciate your help
